spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListenerTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import java.io.Closeable;
+
+import brave.Span;
+import brave.Tracer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Trigger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.cloud.sleuth.instrument.quartz.TracingJobListener.CONTEXT_SPAN_IN_SCOPE_KEY;
+import static org.springframework.cloud.sleuth.instrument.quartz.TracingJobListener.CONTEXT_SPAN_KEY;
+
+/**
+ * @author Branden Cash
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class TracingJobListenerTest {
+
+	@InjectMocks
+	private TracingJobListener listener;
+
+	@Mock
+	private Tracer tracer;
+
+	@Mock
+	private JobExecutionContext context;
+
+	@Mock
+	private JobExecutionException jobException;
+
+	@Mock
+	private Span span;
+
+	@Mock
+	private Closeable spanInScope;
```

Comment:
This is less than ideal ... the `SpanInScope` can't be mocked because its a final class and the constructor is package private. So I went up to the interface I care about, but unfortunately had to the same in `TracingJobListener`. I really don't want to have to shape the code around what is possible in tests. Got any suggestions? Hopefully without wiring up the whole application context.

Reply:
It looks like I may be able to get an actual Tracing/Tracer using something like this:

```
Tracing tracing = Tracing.newBuilder()
		.currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
				.addScopeDecorator(StrictScopeDecorator.create()).build())
		.sampler(Sampler.ALWAYS_SAMPLE).spanReporter(this.reporter).build();

Tracer tracer = this.tracing.tracer();
```

However, i'm still not sure how I'd be able to verify that the `SpanInScope` ever has `close` invoked.

Reply:
Got it figured out by creating a real Tracing object and injecting a mock CurrentTraceContext